devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusBuild.java

```diff
@@ -78,6 +79,30 @@ public FileCollection getClasspath() {
                 .plus(mainSourceSet.getResources());
     }
 
+    @Input
+    public Map<String, Object> getQuarkusBuildSystemProperties() {
+        Map<String, Object> quarkusSystemProperties = new HashMap<>();
+        for (Map.Entry<Object, Object> systemProperty : System.getProperties().entrySet()) {
+            if (systemProperty.getKey() != null && systemProperty.getKey() instanceof String
+                    && systemProperty.getValue() instanceof Serializable
+                    && ((String) systemProperty.getKey()).startsWith("quarkus.")) {
```

Comment:
It's highly unlikely that we'll have a null key (or impossible I'd say), so maybe this can be shortened to: 
```suggestion
            if (systemProperty.getKey().toString().startsWith("quarkus.")
                    && systemProperty.getValue() instanceof Serializable) {
```